jsonlines/jsonlines.py

```diff
@@ -275,15 +273,15 @@ def write(self, obj):
         # the serialised structure can be encoded using ASCII only, so
         # this means this code needs to handle all combinations.
         if self._fp_is_binary:
-            if not isinstance(line, six.binary_type):
+            if not isinstance(line, bytes):
```

Comment:
i think this branch is always taken? in that case the `.encode()` line below can be unconditional.

quick experimentation with `json`, `simplejson`, and `ujson` indicates that they always return `str` (not `bytes`):

```
In [5]: d = {'foo': '🎉'}

In [6]: json.dumps(d)
Out[6]: '{"foo": "\\ud83c\\udf89"}'

In [7]: simplejson.dumps(d)
Out[7]: '{"foo": "\\ud83c\\udf89"}'

In [8]: ujson.dumps(d)
Out[8]: '{"foo":"\\ud83c\\udf89"}'
```